src/forecast.py

```diff
@@ -213,8 +199,100 @@ def forecast_page():
             )
         )
 
-    st.plotly_chart(fig, theme="streamlit")
-
+        # Ensure start_datetimes is not empty
+        if not start_datetimes:
+            st.error("No start dates available. Please check your date inputs.")
+        else:
+            start_datetime_utc = start_datetimes[0]
+            # Check if end_datetimes is empty or the last element is None
+            if not end_datetimes or end_datetimes[-1] is None:
+                # Ensure start_datetime_utc is a datetime object
+                if isinstance(start_datetime_utc, datetime):
+                    # Set end_datetime to 7 days after start_datetime or to current date
+                    end_datetime_utc = max(start_datetime_utc + timedelta(days=7), datetime.utcnow())
+                else:
+                    # If start_datetime_utc is a date object, convert it to datetime
+                    start_datetime_utc = datetime.combine(start_datetime_utc, datetime.min.time())
+                    end_datetime_utc = max(start_datetime_utc + timedelta(days=7), datetime.utcnow())
+            else:
+                end_datetime_utc = end_datetimes[-1]
+                # Ensure end_datetime_utc is a datetime object
+                if isinstance(end_datetime_utc, datetime.date):
+                    end_datetime_utc = datetime.combine(end_datetime_utc, datetime.min.time())
+
```

Comment:
could you refactor this bit to a function, it seems a bit messy in order to get start and end datetimes, but perhaps its the only way